contracts/ERC721A.sol

```diff
@@ -296,7 +341,8 @@ contract ERC721A is IERC721A {
      * Tokens start existing when they are minted (`_mint`),
      */
     function _exists(uint256 tokenId) internal view returns (bool) {
-        return _startTokenId() <= tokenId && tokenId < _currentIndex && !_ownerships[tokenId].burned;
+        return _startTokenId() <= tokenId && tokenId < _currentIndex && // If within bounds,
+            _packedOwnerships[tokenId] & (1 << 224) == 0; // and not burned.
```

Comment:
It might be more readable is we extract `_packedOwnerships[tokenId] & (1 << 224)` into an internal helper function called getBurned or something of that nature. Also makes it easier for inheriting contracts to check if a token is burned

Reply:
Could probably do this for all the fields we used to have on the ownership and address structs

Reply:
Inheriting contracts won't be able to read and write the raw bits.

Reply:
Are you saying they shouldn't be able to or they won't be able to?

Reply:
They shouldn't and won't be able to.